Framework/Kernel/src/GitHubApiHelper.cpp

```diff
@@ -47,12 +51,58 @@ std::string formatRateLimit(const int rateLimit, const int remaining,
       << "Z";
   return msg.str();
 }
+
+/*
+ * Small function to encapsulate getting the token from everything else
+ */
+std::string getApiToken() {
+  // default token is empty string meaning do unauthenticated calls
+  std::string token(DEFAULT_GITHUB_TOKEN);
+  // get the token from configservice if it has been set
+  if (ConfigService::Instance().hasProperty(CONFIG_KEY_GITHUB_TOKEN)) {
+    token = ConfigService::Instance().getString(CONFIG_KEY_GITHUB_TOKEN);
+  }
+
+  // unset is the user's way of intentionally turning of authentication
+  if (token.empty() || token == "unset") {
+    token = "";
+  } else {
+    // error check that token is possibly valid - 40 char
+    // TODO example: 8ec7afc857540ee60af78cba1cf7779a6ed0b6b9
+    if (token.size() != 40) {
+      g_log.notice() << "GitHub API token is not 40 characters (found "
+                     << token.size() << ") with token =\"" << token
+                     << "\" using unauthenticated connection\n";
+      token = "";
+    }
+  }
+
+  // log what the token is and create final string to set in header
+  if (token.empty()) {
+    // only unauthenticated calls
+    g_log.information("Making unauthenticated calls to GitHub");
+    return "";
+  } else {
+    g_log.information("Attempting authenticated calls to GitHub");
+    g_log.debug() << "Using token \"" << token << "\"\n";
```

Comment:
I know this is useful for debugging but do we want to be leaking the token in the logs?

Reply:
I wasn't entirely sure myself. I figured that the average person would never see it, but developers can in case of needing to debug a user's system.